docs/aws/audit/esmonitoring/rules/elasticache_rbac_auth_enabled.mdx

```diff
@@ -23,6 +23,242 @@ CBP
+4. For each Replication Group, check the "Access Control" section. If RBAC Authentication is enabled, it will be listed here. If it is not listed, then RBAC Authentication is not enabled for that Replication Group.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the ElastiCache resources.
+
+2. Once the AWS CLI is set up, you can list all the ElastiCache Replication Groups using the following command:
+
+   ```
+   aws elasticache describe-replication-groups
+   ```
+   This command will return a JSON output with details of all the ElastiCache Replication Groups.
+
+3. To check if RBAC Authentication is enabled for each Replication Group, you need to look for the "AuthTokenEnabled" field in the output. If the value of this field is "false", it means that RBAC Authentication is not enabled.
+
+   You can use the following command to filter the output and list only the Replication Groups where RBAC Authentication is not enabled:
+
+   ```
+   aws elasticache describe-replication-groups --query 'ReplicationGroups[?AuthTokenEnabled==`false`]'
+   ```
+   
+4. If the above command returns any Replication Groups, it means that RBAC Authentication is not enabled for those groups. You should take note of the "ReplicationGroupId" field in the output for further investigation.
+</Accordion>
+
+<Accordion title='Using Python'>
+To check if RBAC Authentication is enabled for ElastiCache Replication Groups in Elastic Search using Python scripts, you can follow these steps:
+
+1. **Set up your AWS SDK for Python (Boto3):**
+   Before you can interact with AWS services, you need to set up your AWS SDK for Python (Boto3). Install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials. You can do this by setting the following environment variables:
+   ```
+   AWS_ACCESS_KEY_ID='your_access_key'
+   AWS_SECRET_ACCESS_KEY='your_secret_key'
+   ```
+
+2. **List all ElastiCache Replication Groups:**
+   Use the `describe_replication_groups()` function to retrieve all ElastiCache Replication Groups. Here's a sample script:
+   ```python
+   import boto3
+
+   def list_replication_groups():
+       client = boto3.client('elasticache')
+       response = client.describe_replication_groups()
+       return response['ReplicationGroups']
+
+   replication_groups = list_replication_groups()
+   print(replication_groups)
+   ```
+   This will print out all the replication groups along with their details.
+
+3. **Check if RBAC is enabled for each Replication Group:**
+   For each replication group, check if the `AuthTokenEnabled` attribute is set to `True`. This indicates that RBAC is enabled. Here's how you can modify the previous script to do this:
+   ```python
+   import boto3
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>